src/app/myProfile/profileUpdate/ProfileUpdate.tsx

```diff
@@ -0,0 +1,138 @@
+import { GymTheme } from '@/MUITheme/GymTheme';
+import '../MyProfile.scss';
+import { Avatar, Button, TextField, ThemeProvider } from '@mui/material';
+import { deepPurple } from '@mui/material/colors';
+import { useForm } from 'react-hook-form';
+import * as Yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+
+export interface Profile {
+  firstname: string;
+  lastname: string;
+  username: string;
+  bio: string;
+  location: string;
+}
+
+interface ProfileUpdateProps {
+  cancelAction: () => void;
+  saveAction: (profile: Profile) => void;
+  profile: Profile;
+}
+
+const validationSchema = Yup.object().shape({
```

Comment:
It shouldnt be in that file. Either yupValidation (or sth like this) or in type file